backend/src/main/java/com/tourneynizer/tourneynizer/controller/TeamRequestController.java

```diff
@@ -0,0 +1,59 @@
+package com.tourneynizer.tourneynizer.controller;
+
+import com.tourneynizer.tourneynizer.error.BadRequestException;
+import com.tourneynizer.tourneynizer.model.ErrorMessage;
+import com.tourneynizer.tourneynizer.model.User;
+import com.tourneynizer.tourneynizer.service.SessionService;
+import com.tourneynizer.tourneynizer.service.TeamRequestService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.CookieValue;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+
+import java.util.Collections;
+
+@Controller("TeamRequestController")
+public class TeamRequestController {
+
+    private final TeamRequestService teamRequestService;
+    private final SessionService sessionService;
+
+    @Autowired
+    public TeamRequestController(TeamRequestService teamRequestService, SessionService sessionService) {
+        this.teamRequestService = teamRequestService;
+        this.sessionService = sessionService;
+    }
+
+
+    @PostMapping("/api/team/{id}/request")
```

Comment:
So only a user can request to join a team?

Reply:
So far yes